ksi-ls12 breaks make

### Expected behavior
When running rsyslog configure  with `--enable-ksi-ls12` it the `make` should work fine
Was working fine in 8.32.0.

### Actual behavior

From buildlog:

```
  CC     lmsig_ksi_ls12_la-lmsig_ksi-ls12.lo
  CC     lmsig_ksi_ls12_la-lib_ksils12.lo
lib_ksils12.c:65: error: redefinition of typedef 'uchar'
typedefs.h:49: note: previous declaration of 'uchar' was here
lib_ksils12.c: In function 'ksiCreateFile':
lib_ksils12.c:557: warning: ignoring return value of 'fwrite', declared with attribute warn_unused_result
make[2]: *** [lmsig_ksi_ls12_la-lib_ksils12.lo] Error 1
make[2]: Leaving directory '/builddir/build/BUILD/rsyslog-8.33.1/runtime'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/builddir/build/BUILD/rsyslog-8.33.1'
make: *** [all] Error 2
```

### Steps to reproduce the behavior

```
./configure --build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --target=x86_64-redhat-linux-gnu --program-prefix= --prefix=/usr --exec-prefix= --bindir=/bin --sbindir=/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/lib64 --libexecdir=/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-static --disable-testbench --enable-uuid --enable-elasticsearch --enable-omkafka --enable-imkafka --enable-kafka-static --enable-usertools --enable-ksi-ls12 --enable-gnutls --enable-imfile --enable-impstats --enable-imptcp --enable-libdbi --enable-mail --enable-mysql --enable-omprog --enable-omudpspoof --enable-omuxsock --enable-omstdout --enable-pgsql --enable-pmlastmsg --enable-relp --enable-snmp --enable-unlimited-select --enable-mmjsonparse --enable-mmnormalize --enable-mmanon --enable-mmutf8fix --enable-mail --enable-mmfields --enable-mmpstrucdata --enable-mmsequence --enable-mmrm1stspace --enable-pmaixforwardedfrom --enable-pmciscoios --disable-liblogging-stdlog
make
```

### Environment.
- rsyslog version: 8.33.1
- platform: CentOS6 / EL6

--- a/runtime/lib_ksi_queue.c
+++ b/runtime/lib_ksi_queue.c
@@ -206,6 +206,7 @@ void *worker_thread_main(void *arg) {
 	WorkerThreadContext* tc = (WorkerThreadContext*) arg;
 
 	while (1) {
+		item = NULL;
 		res = ProtectedQueue_waitForItem(tc->queue, &item, tc->timeout);
 		if (tc->queue->bStop)
 			return NULL;

--- a/runtime/lib_ksi_queue.c
+++ b/runtime/lib_ksi_queue.c
@@ -21,7 +21,7 @@ void RingBuffer_free(RingBuffer* this) {
 
 static bool RingBuffer_grow(RingBuffer* this) {
 	void **pTmp = calloc(this->size * RB_GROW_FACTOR, sizeof (void*));
-	void *pTmpItem;
+	void *pTmpItem = NULL;
 	if (!pTmp)
 		return false;
 
@@ -42,6 +42,9 @@ static bool RingBuffer_grow(RingBuffer* this) {
 bool RingBuffer_pushBack(RingBuffer* this, void* item) {
 
 	if (this->size == this->count && !RingBuffer_grow(this))
+		return false;
+
+	if(this->size == 0)
 		return false;
 
 	this->buffer[this->tail] = item;
@@ -210,7 +213,7 @@ void *worker_thread_main(void *arg) {
 		if (res == ETIMEDOUT) {
 			if (!tc->timeoutFunc())
 				return NULL;
-		} else if (!tc->workerFunc(item))
+		} else if (item != NULL && !tc->workerFunc(item))
 			return NULL;
 	}
 }

--- a/runtime/lib_ksils12.c
+++ b/runtime/lib_ksils12.c
@@ -487,6 +487,9 @@ ksiCloseSigFile(ksifile ksi) {
 
 static int mkpath(char* path, mode_t mode, uid_t uid, gid_t gid) {
 
+	if(path == NULL)
+		return 1;
+
 	for (char *p = strchr(path + 1, '/'); p; p = strchr(p + 1, '/')) {
 		*p = '\0';
 		if (mkdir(path, mode) == 0) {
@@ -514,6 +517,9 @@ ksiCreateFile(rsksictx ctx, const char *path, uid_t uid, gid_t gid, int mode, bo
 	struct stat stat_st;
 	FILE *f = NULL;
 	struct flock lock = {F_WRLCK, SEEK_SET, 0, 0, 0};
+
+	if(path ==NULL)
+		return NULL;
 
 	if (mkpath((char*) path, ctx->fDirCreateMode, ctx->dirUID, ctx->dirGID) != 0) {
 		report(ctx, "ksiCreateFile: mkpath failed for %s", path);
@@ -1243,6 +1249,9 @@ static void process_requests(rsksictx ctx, KSI_CTX *ksi_ctx, FILE* outfile) {
 	int r = KSI_OK;
 	KSI_Signature *sig = NULL;
 
+	if(ProtectedQueue_count(ctx->signer_queue) == 0)
+		return;
+
 	while (ProtectedQueue_peekFront(ctx->signer_queue, (void**) &item) && item->type == QITEM_SIGNATURE_REQUEST) {
 		if (lastItem != NULL) {
 			if (outfile)
@@ -1264,6 +1273,14 @@ static void process_requests(rsksictx ctx, KSI_CTX *ksi_ctx, FILE* outfile) {
 
 	if (!outfile)
 		goto signing_failed;
+
+	if(item->type != QITEM_SIGNATURE_REQUEST)
+		return;
+
+	if(lastItem->arg == NULL) {
+		r = KSI_INVALID_ARGUMENT;
+		goto signing_failed;
+	}
 
 	r = KSI_Signature_signAggregated(ksi_ctx, lastItem->arg, lastItem->intarg2, &sig);
 	if (r != KSI_OK) {

--- a/runtime/lib_ksils12.c
+++ b/runtime/lib_ksils12.c
@@ -62,7 +62,6 @@
 #include "lib_ksils12.h"
 #include "lib_ksi_queue.h"
 
-typedef unsigned char uchar;
 #ifndef VERSION
 #define VERSION "no-version"
 #endif
@@ -553,8 +552,13 @@ ksiCreateFile(rsksictx ctx, const char *path, uid_t uid, gid_t gid, int mode, bo
 		goto done;
 	}
 
-	if (stat_st.st_size == 0 && header != NULL)
-		fwrite(header, strlen(header), 1, f);
+	if (stat_st.st_size == 0 && header != NULL) {
+		if(fwrite(header, strlen(header), 1, f) != 1) {
+        	        report(ctx, "ksiOpenSigFile: fwrite for file %s failed: %s",
+				path, strerror(errno));
+	                goto done;
+		}
+	}
 
 done:
 	return f;

--- a/runtime/lib_ksils12.c
+++ b/runtime/lib_ksils12.c
@@ -1274,7 +1274,7 @@ static void process_requests(rsksictx ctx, KSI_CTX *ksi_ctx, FILE* outfile) {
 	if (!outfile)
 		goto signing_failed;
 
-	if(item->type != QITEM_SIGNATURE_REQUEST)
+	if(!lastItem || item->type != QITEM_SIGNATURE_REQUEST)
 		return;
 
 	if(lastItem->arg == NULL) {

